From 51df997cdf38dbb5381e136b65460e01d00bbd55 Mon Sep 17 00:00:00 2001
From: Rob Evers <revers@redhat.com>
Date: Fri, 27 Jun 2014 14:29:34 -0400
Subject: [scsi] be2iscsi: Jump to 'free_memory' is apparently missing

Message-id: <1403879375-31620-1-git-send-email-revers@redhat.com>
Patchwork-id: 85056
O-Subject: [RHEL7.1 e-stor PATCH 07/10] scsi: be2iscsi: add an missing goto in error path
Bugzilla: 1088576
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>

From: John Soni Jose <sony.john-n@emulex.com>

a jump to 'free_memory' is apparently missing

http://marc.info/?l=linux-scsi&m=140205639608350&w=2

Signed-off-by: Tomas Henzl <thenzl@redhat.com>
Reviewed-by: Mike Christie <michaelc@cs.wisc.edu>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/be2iscsi/be_main.c b/drivers/scsi/be2iscsi/be_main.c
index 57605ef..f2f034b 100644
--- a/drivers/scsi/be2iscsi/be_main.c
+++ b/drivers/scsi/be2iscsi/be_main.c
@@ -4248,6 +4248,8 @@ static int hba_setup_cid_tbls(struct beiscsi_hba *phba)
   kfree(phba->ep_array);
   phba->ep_array = NULL;
   ret = -ENOMEM;
+
+  goto free_memory;
  }
 
  for (i = 0; i < phba->params.cxns_per_ctrl; i++) {
-- 
1.7.1